lib/sidekiq/cron/poller.rb

```diff
@@ -45,6 +42,10 @@ def enqueue_job(job, time = Time.now.utc)
         Sidekiq.logger.error "CRON JOB: #{ex.backtrace.first}"
         handle_exception(ex) if respond_to?(:handle_exception)
       end
+
+      def poll_interval_average
+        @config[:average_scheduled_poll_interval] || POLL_INTERVAL
```

Comment:
I do not agree with using `average_scheduled_poll_interval` here because this makes it intertwined with the Sidekiq setting.

Default value for this in Sidekiq is actually 5 seconds. So when we introduced this option, we actually dropped the default interval to 5 seconds and the docs are wrong.

I think a better approach would be to use a different setting like `cron_poll_interval`. I kept this as-is in this PR though since this is a different bug.

Reply:
I agree with this, we should not change the Sidekiq settings... this will be surprising for most of the users and can potentially introduce weird bugs. It's ok to read Sidekiq settings to take, for example, default settings... but overriding those settings seems a bad idea.

Reply:
Created a separate issue for this: https://github.com/ondrejbartas/sidekiq-cron/issues/361

Reply:
btw, where this method is called?

Reply:
This is called in the parent class: https://github.com/mperham/sidekiq/blob/v6.4.2/lib/sidekiq/scheduled.rb#L150-L157